### DISCUSSION TITLE
Trying to replicate my old vim setup in neovim

### DISCUSSION THREAD

**Discussion by statquant** [R-nvim/R.nvim#129] [Q&A]
Hello, I am in the process of migrating from vim to nvim and I can't figure out how to do the following.
Is a discussion the right place to ask for guidance ?

1. When I start `R` with `<leader>rf` I used to get the R console in a tmux split pane, is that still possible ?
    I'm guessing not because in vim it was already deprecated, in that case would that be possible to always switch in insert mode when going to terminal window ? 
3. Is it possible to map `<C-CR>` to SendLine in insert mode while staying on the same position, I managed to map `<Leader><CR>` but he cursor go to next line ?
4. To get completion I need to start a R session is that expected ?
5. I cannot seem to be able to navigate the choice, how should I do it ?
![image](https://github.com/R-nvim/R.nvim/assets/3612473/7b98b9a0-2b9f-4e96-a49f-da77c4e95ced)

Thank you very much for you plugin

**Comment by jalvesaq** [comment]
This is the right place for your questions.

1. Starting R in a Tmux split pane wasn't ported to `R.nvim`. An alternative to this was discussed at https://github.com/R-nvim/R.nvim/issues/115 and someone interested in reimplementing and maintaining the feature can do a pull request.

2. Key bindings can be configured in the function created as a `hook.on_filetype` option. Please look at the example in `R.nvim` documentation, section _Custom key bindings_ (`R.nvim-key-bindings`).

3. You don't need to start R to get the completion of objects from installed packages if the libraries were loaded at least once in an R session started with `<LocalLeader>rf` or if the corresponding `library()` commands were already present in the script when you started to edit it. These are the two circumstances when the cache files for completion are built.

4. [nvim-cmp](https://github.com/hrsh7th/nvim-cmp) doesn't have pre-defined key bindings. You have to configure them. Please, look at the examples on the [Wiki](https://github.com/R-nvim/R.nvim/wiki/Configuration).

**Comment by statquant** [comment]
Thank you for all your pointers.

**Comment by jalvesaq** [comment]
Update:
1. Tmux split is now supported. The documentation on `external_term` has an example of how to configure it.
2. I realized that it's better to create key bindings for `<Plug>` labels in the `init.vim` or `init.lua`, before loading the plugin.
